### PR TITLE
[search] Fixed numero signs in search queries and names.

### PR DESCRIPTION
**Pull request by ygorshenin**
'#' and '№' before digits are considered as spaces now.

### CODE REVIEW

**Comment by mpimenov** [comment]
https://jira.mail.ru/browse/MAPSME-4586

File: indexer/search_string_utils.cpp [hunk 1]
```diff
@@ -22,41 +51,54 @@ UniString NormalizeAndSimplifyString(string const & s)
...
+      break;
     // Some Danish-specific hacks.
-    case 0x00d8:                    // Ø
-    case 0x00f8: c = 'o'; break;    // ø
-    case 0x0152:                    // Œ
-    case 0x0153:                    // œ
+    case 0x00d8:  // Ø
+    case 0x00f8:
+      c = 'o';
+      break;      // ø
```
Comment (kshalnev) [inline indexer/search_string_utils.cpp]: м.б. оставить комментарий на строке 71: 
```
case 0x00f8: // ø
```

File: indexer/search_string_utils.hpp [hunk 2]
```diff
@@ -1,63 +1,69 @@
...
 
 // This function should be used for all search strings normalization.
 // It does some magic text transformation which greatly helps us to improve our search.
-strings::UniString NormalizeAndSimplifyString(string const & s);
+strings::UniString NormalizeAndSimplifyString(std::string const & s);
 
-template <class DelimsT, typename F>
-void SplitUniString(strings::UniString const & uniS, F f, DelimsT const & delims)
+template <class Delims, typename Fn>
+void SplitUniString(strings::UniString const & uniS, Fn f, Delims const & delims)
```
Comment (kshalnev) [inline indexer/search_string_utils.hpp]: м.б. Fn&&?

File: indexer/search_string_utils.cpp (hunk 1 above)
Comment (ygorshenin) [inline indexer/search_string_utils.cpp]: Done

File: indexer/search_string_utils.hpp (hunk 2 above)
Comment (ygorshenin) [inline indexer/search_string_utils.hpp]: No real need now.

**Comment by ygorshenin** [comment]
@mpimenov @kshalnev PTAL

**Pull request merged**